### DISCUSSION TITLE
potential issue with tsx watch

### DISCUSSION THREAD

**Discussion by onemen** [epicweb-dev/epic-stack#103] [Q&A]
We have this set up to run in development

```json
    "dev": "remix dev -c \"npm run dev:server\" --no-restart",
    "dev:server": "cross-env MOCKS=true tsx watch --clear-screen=false --ignore \"app/**\" --ignore \"build/**\" --ignore \"node_modules/**\" ./index.js",
```

Currently everything work without any issue.
However as soon as I add new ESM package and import it in the server code, the app stop responding at the import line. 
i've tested :
- @mdx-js/esbuild
- chalk 5.2.0
- execa
- globby

For example, `epic-stack` current version of `chalk` which is used in `server/index.ts` is cjs, if you install the latest version (Chalk 5 is ESM), the server will not start at all, without changing `dev:server` script as mention below.

I've see this on Windows 11.
Can you all test it on your platform and report if you can reproduce this issue and if the script below fix it.

Changing `dev:server` to this script fix the issue
```json
"dev:server": "cross-env MOCKS=true tsx ./index.js",
```
When we get more info on this we can report it to remix

**Comment by onemen** [comment]
@kentcdodds did you check this?

**Comment by kentcdodds** [comment]
Hmmm... This is surprising to me. Removing the `watch` command and the flags are all there for a reason. When a change is made to the `server/index.ts`, the server needs to be restarted. But we want to make sure to not restart the server when other files are changed. Can we dig a little more to understand why these flags cause this issue?

**Comment by onemen** [comment]
can you test how how it work on macos?, maybe it is only a windows issue?

**Comment by onemen** [comment]
I've tested this issue on my virtual machines, and i can confirm that this is Windows issue only.

updating this code fix the issue
https://github.com/remix-run/remix/blob/761fc1d867b674e8c00268c263675fe04f43a10a/packages/remix-dev/devServer_unstable/index.ts#L69-L79
```diff
    let newAppServer = execa.command(command, {
-     stdio: "pipe",
+     stdio: ['ignore', 'pipe', 'pipe'],
+     shell: true,
      env: {
        NODE_ENV: "development",
        PATH:
          bin + (process.platform === "win32" ? ";" : ":") + process.env.PATH,
        REMIX_DEV_HTTP_ORIGIN: stringifyOrigin(httpOrigin),
      },
      // https://github.com/sindresorhus/execa/issues/433
      windowsHide: false,
    });
```

I have report about it to Remix [#6504](https://github.com/remix-run/remix/issues/6504)

**Reply by xHomu** [reply]
Can confirm this fixed my issue on Windows. 

Nice find!
